compiler/plugins/target/ROCM/builtins/ukernel/iree_uk_amdgpu_multi_mma_mfma_i32_16x16x32_i8.c

```diff
@@ -6,46 +6,176 @@
 
 #include "compiler/plugins/target/ROCM/builtins/ukernel/common.h"
 
-// Very naive kernel. TODO(bjacob):
-// 1. Shared memory: can't allocate it within the microkernel (which is just a
-//    helper device function, not the actual amdgpu_kernel). Need to get it
-//    passed down here as additional parameters.
-// 2. Better scheduling via either barrier intrinsics or inline assemby.
-[[clang::always_inline]] void iree_uk_amdgpu_multi_mma_mfma_i32_16x16x32_i8(
-    const int8_t *a_buffer, int64_t a_offset, const int8_t *b_buffer,
-    int64_t b_offset, int32_t *c_buffer, int64_t c_offset, int32_t k_size,
+// Encodes some information about a A/B fragment tile;
+typedef struct ab_tile_info_t {
+  // Terminology: "_vecs":
+  // We will be counting in units of "vectors", meaning, for each A/B fragment
+  // the corresponding operand type of the MFMA intrinsic. For A and B, that
+  // type is i64, used as <8 x i8>.
```

Comment:
I don't see how we jumped straight to i64? The fundamental data type on this hw is dword == i32. I'd either stick with dwords or spell out that this is the case for 16x16x32_i8 specifically.

Reply:
The comment was already saying "the corresponding operand type of the MFMA intrinsic", maybe I can change `s/the/this particular/` ?